app/services/post_scraper.rb

```diff
@@ -143,179 +109,80 @@ def threaded_page_links
 
   def import_post_from_doc(doc)
     subject = @subject || doc.at_css('.entry .entry-title').text.strip
-    logger.info "Importing thread '#{subject}'"
+    Resque.logger.info "Importing thread '#{subject}'"
 
     username = doc.at_css('.entry-poster b').inner_html
-    img_url = doc.at_css('.entry .userpic img').try(:attribute, 'src').try(:value)
-    img_keyword = doc.at_css('.entry .userpic img').try(:attribute, 'title').try(:value)
+    img_node = doc.at_css('.entry .userpic img')
+    img_url = img_node.try(:attribute, 'src').try(:value)
+    img_keyword = img_node.try(:attribute, 'title').try(:value)
     created_at = doc.at_css('.entry .datetime').text
     content = doc.at_css('.entry-content').inner_html
 
-    @post = Post.new
-    @post.board_id = @board_id
-    @post.section_id = @section_id
-    @post.subject = subject
-    @post.content = strip_content(content)
-    @post.created_at = @post.updated_at = @post.edited_at = created_at
-    @post.status = @status
-    @post.is_import = true
-
     # detect already imported
     # skip if it's a threaded import, unless a subject was given manually
-    if (@subject || !@threaded_import) && (subj_post = Post.where(subject: @post.subject, board_id: @board_id).first)
-      raise AlreadyImportedError.new("This thread has already been imported", subj_post.id)
+    if (@subject || !@threaded_import) && (subj_post = Post.find_by(subject: subject, board_id: @board_id))
+      @errors.add(:post, "was already imported! #{ScrapePostJob.view_post(subj_post.id)}")
+      raise ActiveRecord::Rollback
     end
 
-    set_from_username(@post, username)
-    @post.last_user_id = @post.user_id
-
-    set_from_icon(@post, img_url, img_keyword)
+    @post = Post.new(board_id: @board_id, section_id: @section_id, subject: subject, status: @status, is_import: true)
```

Comment:
note for me to check: Do we set `created_at`, `updated_at`, `edited_at` in ReplyScraper?

Reply:
We do!

Reply:
Yes :P